File: 단계별로 풀어보기/15. 약수, 배수와 소수 2/13241.cpp
#include <iostream>
using namespace std;

int Euclidean(int a, int b)
{
	if (b == 0)
		return a;
	else if (b == 1)
		return 1;
	
	return Euclidean(b, a % b);
}

int main()
{
	long long num1, num2, LCV, result;
	cin >> num1 >> num2;
	LCV = (num1 > num2) ? Euclidean(num1, num2) : Euclidean(num2, num1);
	result = num1 * num2 / LCV;
	cout << result;
	return 0;
}

File: step by step/15. Measure and Drainage/13241.py
def Euclidean(a, b):
    if b == 0:
        return a
    elif b == 1:
        return 1
    return Euclidean(b, a % b)


num1, num2 = map(int, input().split())
LCV = Euclidean(num1, num2) if num1 > num2 else Euclidean(num2, num1)
result = num1 * num2 // LCV
print(result)
